chalice/package.py

```diff
@@ -357,15 +386,255 @@ def _register_cfn_resource_name(self, name):
         return cfn_name
 
 
+class TerraformGenerator(TemplateGenerator):
+
+    template_file = "chalice.tf.json"
+
+    def generate(self, resources):
+        # type: (List[models.Model]) -> Dict[str, Any]
+        template = {
+            'resource': {},
+            'terraform': {
+                'required_version': '> 0.11.0'
+            },
+            'data': {
+                'aws_caller_identity': {'chalice': {}},
+                'aws_region': {'chalice': {}}
+            }
+        }
+        for resource in resources:
+            self.dispatch(resource, template)
+        return template
+
+    def _fref(self, lambda_function, attr='arn'):
+        # type: (models.ManagedModel, str) -> str
+        return '${aws_lambda_function.%s.%s}' % (
+            lambda_function.resource_name, attr)
+
+    def _arnref(self, arn_template, **kw):
+        # type: (str, str) -> str
+        d = dict(
+            region='${aws_region.chalice.name}',
+            account_id='${aws_caller_identity.chalice.account_id}')
+        d.update(kw)
+        return arn_template % d
+
+    def _generate_managediamrole(self, resource, template):
+        # type: (models.ManagedIAMRole, Dict[str, Any]) -> None
+        template['resource'].setdefault('aws_iam_role', {})[
+            resource.resource_name] = {
+                'name': resource.role_name,
+                'assume_role_policy': json.dumps(resource.trust_policy)
+        }
+
+        template['resource'].setdefault('aws_iam_role_policy', {})[
+            resource.resource_name] = {
+                'name': resource.resource_name + 'Policy',
+                'policy': json.dumps(resource.policy.document),
+                'role': '${aws_iam_role.%s.id}' % resource.resource_name,
+        }
+
+    def _generate_s3bucketnotification(self, resource, template):
+        # type: (models.S3BucketNotification, Dict[str, Any]) -> None
+
+        bnotify = {
+            'events': resource.events,
+            'lambda_function_arn': self._fref(resource.lambda_function)
+        }
+
+        if resource.prefix:
+            bnotify['filter_prefix'] = resource.prefix
+        if resource.suffix:
+            bnotify['filter_suffix'] = resource.suffix
+
+        template['resource'].setdefault('aws_s3_bucket_notification', {})[
+            resource.resource_name] = {
+                'bucket': resource.bucket,
+                'lambda_function': bnotify
+        }
+
+        template['resource'].setdefault('aws_lambda_permission', {})[
```

Comment:
So ran into [this gotcha](https://www.terraform.io/docs/providers/aws/r/s3_bucket_notification.html) when converting my app. Specifically we can only one bucket notification resource per bucket and in my app, I have two lambda function subscribing to the same bucket. I think to support this we will have to consolidate all lambda functions to the same bucket notification if they point to the same bucket. Otherwise, you get in this weird state where only one of the notification configurations will get applied.

Reply:
thats an inherent limitation of s3 (no overlapped notifications by prefix), not a tooling issue. frankly for any real world usage, i would expect the notifications to come in via sns or trail data events, bucket notify direct to lambda/sqs is a neat trick but its single purpose/usage.

Reply:
I think I misunderstood, are both of those lambda functions in the chalice app?  I can add support for that.

Reply:
added in support for doing multiple function in the same chalice app.

Reply:
>> Are both of those lambda functions in the chalice app?

Yep. I ran into that when porting this app: https://github.com/kyleknap/chalice-workshop/commit/73bcfecf24bebcbdda910a6e9ff6f78fe0cc95ec. I had one lambda function that ran on uploads and another that ran on deletes. The update looks good.